src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java

```diff
@@ -196,6 +196,44 @@ public enum IsaacUserPreferences {
 
   public static final Integer DO_HASHCODE_PRIME = 31;
 
+  // Email Template Tokens
```

Comment:
I know you're doing this because you're replicating the style that's already present in the project, but having a file where you have all your constants can be considered an antipattern, because it reduces modularity. Ideally all these constants should live in the class where they belong. For instance, all these are only used in `EventBookingManager`, so I would have put them there, with the right visibility (`private`, probably). This way your code is more modular and less coupled.

Having said that, I'm not sure if it's worth moving them, because the problem would be that you'd have two different styles: some variables are stored in this `Constants` file, and some others in their respective file. So feel free to leave them here, but please consider this for the future.

Reply:
I originally had the newly added string constants in their associated files but Charles suggested I put them in the central constants file with the rest instead.
I'm fine with moving them back but it might need discussion between you two so we don't end up with things moving back and forth depending who is doing the reviews that week? 🤷 
I believe this should be the relevant conversation if you want to review it yourself: https://github.com/isaaccomputerscience/isaac-api/pull/156#discussion_r1523628711

Reply:
@jmgq from a readability/maintainability point of view I thought text constants made more sense in a single location. Functional constants, absolutely yes keep them close to where they're used - if you need to change them you'll be looking at that code. But if all you want to do is change the content of a string displayed in the UI, IMO it makes more sense to have them in a more user friendly place.